include/aws/common/aws_profile.h

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#ifndef AWS_COMMON_AWS_PROFILE_H
+#define AWS_COMMON_AWS_PROFILE_H
+
+#include <aws/common/common.h>
+#include <aws/common/hash_table.h>
+
+struct aws_allocator;
+struct aws_string;
+struct aws_byte_buf;
+struct aws_byte_cursor;
+
+/*
+ * A set of data types that model the aws profile specification
+ *
+ * A profile collection is a collection of zero or more named profiles
+ * Each profile is a set of properties (named key-value pairs)
+ * Empty-valued properties may have sub properties (named key-value pairs)
+ *
+ * Resolution rules exist to determine what profile to use, what files to
+ * read profile collections from, and what types of credentials have priority.
+ *
+ * The profile specification is informally defined as "what the aws cli does" and
+ * formally defined in internal aws documents.
+ */
+struct aws_profile_property {
+    struct aws_allocator *allocator;
+    struct aws_string *name;
+    struct aws_string *value;
+    struct aws_hash_table sub_properties;
+    bool is_empty_valued;
+};
+
+struct aws_profile {
+    struct aws_allocator *allocator;
+    struct aws_string *name;
+    struct aws_hash_table properties;
+    bool has_profile_prefix;
+};
+
+/**
+ * The profile specification has rule exceptions based on what file
+ * the profile collection comes from.
+ */
+enum aws_profile_source_type { AWS_PST_NONE, AWS_PST_CONFIG, AWS_PST_CREDENTIALS };
+
+struct aws_profile_collection {
+    struct aws_allocator *allocator;
+    enum aws_profile_source_type profile_source;
+    struct aws_hash_table profiles;
+};
```

Comment:
move this definition to the .c file
no benefit in having it here, since it has new()/destroy() semantics